deepchecks/vision/datasets/classification/simple.py

```diff
@@ -0,0 +1,205 @@
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module contains implementation of the simple classification dataset."""
+import typing as t
+from pathlib import Path
+
+import cv2
+import PIL.Image as pilimage
+import torch
+import numpy as np
+from torch.utils.data import DataLoader
+from torchvision.datasets import VisionDataset
+from typing_extensions import Literal
+
+from deepchecks import vision
+
+
+__all__ = ['load_dataset', 'SimpleClassificationDataset']
+
+
+def load_dataset(
+    root: str,
+    train: bool = True,
+    batch_size: int = 32,
+    num_workers: int = 0,
+    shuffle: bool = True,
+    pin_memory: bool = True,
+    object_type: Literal['VisionData', 'DataLoader'] = 'DataLoader',
+    **kwargs
+) -> t.Union[DataLoader, vision.ClassificationData]:
+    """Load simple classification dataset.
+
+    The function expects that data within root folder
+    will be structured in a next way:
+
+        - root/
+            - train/
+                - class1/
+                    image1.jpeg
+            - test/
+                - class1/
+                    image1.jpeg
+
+    Parameters
+    ----------
+    root : str
+        path to the data
+    train : bool
+        if `True` train dataset, otherwise test dataset
+    batch_size : int, default: 32
+        Batch size for the dataloader.
+    num_workers : int, default: 0
+        Number of workers for the dataloader.
+    shuffle : bool, default: True
+        Whether to shuffle the dataset.
+    pin_memory : bool, default: True
+        If ``True``, the data loader will copy Tensors
+        into CUDA pinned memory before returning them.
+    object_type : Literal['Dataset', 'DataLoader'], default: 'DataLoader'
+        type of the return value. If 'Dataset', :obj:`deepchecks.vision.VisionDataset`
+        will be returned, otherwise :obj:`torch.utils.data.DataLoader`
+
+    Returns
+    -------
+    Union[DataLoader, VisionDataset]
+
+        A DataLoader or VisionDataset instance representing COCO128 dataset
+    """
+
+    def batch_collate(batch):
+        imgs, labels = zip(*batch)
+        return list(imgs), list(labels)
+
+    dataloader = DataLoader(
+        dataset=SimpleClassificationDataset(root=root, train=train, **kwargs),
+        batch_size=batch_size,
+        shuffle=shuffle,
+        num_workers=num_workers,
+        collate_fn=batch_collate,
+        pin_memory=pin_memory,
+        generator=torch.Generator()
+    )
+
+    if object_type == 'DataLoader':
+        return dataloader
+    elif object_type == 'VisionData':
+        return SimpleClassificationData(data_loader=dataloader)
+    else:
+        raise TypeError(f'Unknown value of object_type - {object_type}')
+
+
+class SimpleClassificationDataset(VisionDataset):
+    """Simple VisionDataset type for the classification tasks.
+
+    Current class expects that data within root folder
```

Comment:
```suggestion
    The current class expects that data within the root folder
```
Apart from that, update also this docstrings according to comments above.

Reply:
done